src/expr/src/scalar/mod.rs

```diff
@@ -86,7 +87,9 @@ impl Arbitrary for MirScalarExpr {
             (0..10)
                 .prop_map(|i| MirScalarExpr::Column(i as usize))
                 .boxed(),
-            // TODO (#12125): add a leaf variant for MirScalarExpr::literal_ok
+            (arb_datum(), any::<ScalarType>())
```

Comment:
Don't know if this is worth the effort, but we could put a TODO here saying that ideally we would pick an arbitrary datum after we pick an arbitrary `ScalarType` because only some kinds of data are allowed for each `ScalarType`.

Reply:
I added a TODO on `arb_datum()` that we need a generation strategy that is parameterized by a `ColumnType` and created #12354 to track this.